### PR TITLE
Update Organization Backend SDK docs to match BAPI docs

### PR DESCRIPTION
**Pull request by alexisintech**
> [!IMPORTANT]
> 🔎 Previews:
>
> -

### Explanation:

Part of the work for checking that our /backend/* reference docs were up to date ([due to this linear ticket](https://linear.app/clerk/issue/DOCS-6039/check-that-our-docsreferencesbackend-pages-match-our-backend-api))

I added each pages' corresponding `clerk/javascript` file in a comment on at the top of the page. Adding these links will make it easy to find where these methods exist in the `clerk/javascript` repo when updating their respective pages.

I updated pages accordingly to reflect the BAPI docs.

### CODE REVIEW

File: docs/references/backend/organization/update-organization-metadata.mdx [hunk 1]
```diff
@@ -3,9 +3,11 @@ title: updateOrganizationMetadata()
 description: Use Clerk's Backend SDK to update the metadata associated with a given organization.
 ---
 
+{/* clerk/javascript file: https://github.com/clerk/javascript/blob/main/packages/backend/src/api/endpoints/OrganizationApi.ts#L159 */}
+
 # `updateOrganizationMetadata()`
 
-Updates the metadata associated with the specified organization ID.
+Updates the metadata attributes of an [`Organization`](/docs/references/backend/types/backend-organization) by merging existing values with the provided parameters. Metadata values will be updated via a "deep" merge - "deep" meaning that any nested JSON objects will be merged as well. You can remove metadata keys at any level by setting their value to `null`.
```
Comment (JacobMGEvans) [inline docs/references/backend/organization/update-organization-metadata.mdx:10]: I thought I was rereading the same changes and realized its just the same message in another section 😅 Always good to have the information consistent!

**Review by JacobMGEvans — APPROVED** [review]
Improvements and additions to Orgs Backend look solid!

File: docs/references/backend/organization/get-organization-invitation.mdx [hunk 2]
```diff
@@ -3,9 +3,11 @@ title: getOrganizationInvitation()
 description: Use Clerk's Backend SDK to retrieve an organization invitation.
```
Comment (royanger) [inline docs/references/backend/organization/get-organization-invitation.mdx:3]: This page isn't in the manifest.

Comment (alexisintech) [reply docs/references/backend/organization/get-organization-invitation.mdx:3]: handled [here](https://github.com/clerk/clerk-docs/pull/1253) - please review

**Pull request merged**